pkg/model/mask.go

```diff
@@ -115,6 +110,50 @@ func updateField(src, dst reflect.Value, fields []string) {
 	}
 }
 
+// updateSliceField this function updates a destination slice by either replacing a struct with a matching ID
+// or appending new values if no matching ID is found.
+func updateSliceField(srcField, dstField reflect.Value) {
+	appendOnly := func() bool {
+		el := srcField.Type().Elem()
+
+		if el.Kind() != reflect.Struct {
+			return true
+		}
+
+		_, ok := el.FieldByName("ID")
+
+		return !ok
```

Comment:
A litle more simple
```suggestion
appendOnly := func() bool {
    el := srcField.Type().Elem()
    _, ok := el.FieldByName("ID")
     
     return el.Kind() != reflect.Struct || !ok
    }
```

Reply:
This solution will cause a panic. We should ensure that the type on which we are calling FieldByName is indeed a struct type.

Reply:
oki

Reply:
It is less performant, too.

Reply:
@VladislavsPerkanuks I like the thinking, though. But check my previous comment. :)